src/main/java/io/nv/bukkit/CleanroomGenerator/CleanroomChunkGenerator.java

```diff
@@ -52,8 +54,20 @@ public CleanroomChunkGenerator() {
             }
 
             String tokens[];
+            String parts[];
 
-            tokens = id.split("[|]");
+            parts = id.split("!");
+            tokens = parts[0].split("[|]");
+
+            try {
+                worldBiome = Biome.valueOf(parts[1].toUpperCase());
```

Comment:
I imagine this will throw an index out of range exception if no biome is specified, which is then caught, but still rather icky. Can you add a length check on parts before trying to set the biome? At which point the second try-catch around the log.warning should no longer be needed too.